hips/hip-9999.md

```diff
@@ -0,0 +1,254 @@
+
+Another common motivation for this HIP is to indicate that database migration output running as a hook should be visible to the user in the case of failure.
+
+In general, allowing chart developers to run jobs and present that feedback directly to the users could also open up additional use cases beyond just the preflight use case that motivated this HIP.  I could imagine scenarios where maybe CVE warnings are presented or specific upgrade feedback is presented instead of just a Helm install failure.
+
+## Rationale
+
+There are other ways that this could be implemented. For example, we could have a separate preflight hook type. However, this new hook type wouldn't be handled at all by previous versions of Helm. With this design, it requires minimal changes to Helm and allows for backwards compatibility.
+
+Another strategy could be for Helm to include Troubleshoot.sh as a dependent library, but this could result in too tight of a coupling between the projects and lower overall flexibility and adaptability. 
+
+## Specification
+
+Templates may include the following annotations on Jobs or Pods:
+    
+```yaml
+        "helm.sh/hook": pre-install, pre-upgrade
+        "helm.sh/hook-output-log-policy": hook-failed # or hook-succeeded or hook-failed,hook-succeeded
+```
+
+
+`helm.sh/hook-output-log-policy` would indicate that Helm should display the output of the Job to the user.
+
+Additionally, a new user flag should be created `--no-log-output`  that would skip the output of logs.
+
+Additionally, there will be a new item added to the action SDK configuration to allow SDK consumers to get the output. 
+By default this output will be written to stdout, but an SDK consumer can overwrite the HookOutputFunc to provide a custom writer.
+
+
+```go
+type Configuration struct {
+  ...
+	// Called with container name and returns and expects writer that will receive the log output
+	HookOutputFunc func(namespace, pod, container string) io.Writer
+}
+```
+
+## Backwards compatibility
+
+The only backwards compatibility concern would be that scripts parsing `helm install` output would see some additional text in the case of logs being output. The fact that notes already make the output unstructured should mitigate any concern here. Since we already are trusting chart developers to provide output in the form of notes, this is a logical extension of that that allows the developer to provide more dynamic output. 
```

Comment:
```suggestion
The only backwards compatibility concern would be that scripts parsing `helm install` output would see some additional text in the case of logs being output. The fact that notes already make the output unstructured should mitigate any concern here. Since we already are trusting chart developers to provide output in the form of notes, this is a logical extension of that that allows the developer to provide more dynamic output. The logs would be written to `stderr`, further mitigating any breaking changes to scripts using `helm` CLI.
```

Reply:
@z4ce I added a some suggestions in the address Matt's comments. If you are happy with them, feel free to accept them so we can get a re-review of the PR